runtime/onert/backend/trix/ops/BulkLayer.cc

```diff
@@ -49,24 +50,57 @@ void BulkLayer::configure(const std::vector<const IPortableTensor *> &inputs,
     throw std::runtime_error("Unable to extract the model metadata");
   }
 
+  _model_id.resize(_dev_context->getDevSize());
+
   generic_buffer model_file;
   model_file.type = BUFFER_FILE;
   model_file.filepath = binary_path.c_str();
   model_file.size = _meta->size;
 
-  if (registerNPUmodel(dev_context->getDev(), &model_file, &_model_id) < 0)
+  if (registerNPUmodel(dev_context->getDev(0), &model_file, &_model_id[0]) < 0)
+  {
+    throw std::runtime_error("Failed to register npu model");
+  }
+  if (registerNPUmodel(dev_context->getDev(1), &model_file, &_model_id[1]) < 0)
   {
     throw std::runtime_error("Failed to register npu model");
   }
 }
 
+void single_job(npudev_h dev, int req_id, input_buffers *input_buf, tensors_data_info *in_info,
+                output_buffers *output_buf, tensors_data_info *out_info)
+{
+  if (setNPU_requestData(dev, req_id, input_buf, in_info, output_buf, out_info))
+  {
+    throw std::runtime_error("Unable to create NPU request for red_id (" + std::to_string(req_id) +
+                             ")");
+  }
+
+  if (submitNPU_request(dev, req_id))
+  {
+    throw std::runtime_error("Unable to submit NPU request with req id (" + std::to_string(req_id) +
+                             ")");
+  }
+}
+
 void BulkLayer::run()
 {
-  int req_id;
-  if (createNPU_request(_dev_context->getDev(), _model_id, &req_id))
+  // TODO: Remove too many assumption
+  // We assume user wants batch execurion if user's input size is larger than model's input size
+  int user_input_batch = (_inputs[0]->get_info().shape()).dim(0);
+  int model_input_batch = _meta->input_seg_dims[0][0];
+  int batch_size = user_input_batch / model_input_batch;
+  bool is_batch_execution = (batch_size != 1 ? true : false);
```

Comment:
@chunseoklee Can `is_batch_execution` be determined in `configure` stage, not in `run`? I think you're thinking dynamic shape. But I am not sure multiple batch would work for dynamic shape. For example, AFAIK, trix is not ready for dynamic shape.

Reply:
This calculation works for both static and dynamic shape cases even though we do not have dynamic shape inference for Bulk.

Reply:
Then, I prefer to move this code in `configure` and use `is_batch_execution` to have single batch and multiple match different (its own way) for readability and removing redundant code. But it is just my preference. I am okay for current code.

Reply:
(optional) `is_batch_execution` determined by both `batch_size` and `_dev_context->getDevSize()`. Also, it can be determined in `configure`, not each `run` since we don't have plan to support dynamic shape yet.